File: src/util/logger.ts
import { c, sym } from "./colors.js";
let verbose = false;
let quiet = false;

export function setVerbose(v: boolean) {
  verbose = v;
  if (v) quiet = false;
}
export function setQuiet(q: boolean) {
  quiet = q;
}

export const log = {
  info: (msg: string) => console.log(`${c.cyan(sym.info)} ${msg}`),
  ok: (msg: string) => console.log(`${c.green(sym.check)} ${msg}`),
  warn: (msg: string) => console.log(`${c.yellow(sym.warn)} ${msg}`),
  err: (msg: string) => console.error(`${c.red(sym.cross)} ${msg}`),
  step: (msg: string) => {
    if (!quiet) console.log(`\n${c.bold(c.magenta(sym.arrow + " " + msg))}`);
  },
  sub: (msg: string) => {
    if (!quiet) console.log(`  ${c.gray(sym.bullet)} ${msg}`);
  },
  debug: (msg: string) => {
    if (verbose) console.log(`  ${c.gray("[debug] " + msg)}`);
  },
  raw: (msg: string) => console.log(msg),
  banner: (title: string, subtitle?: string) => {
    if (quiet) return;
    const line = "─".repeat(Math.max(title.length, subtitle?.length ?? 0) + 4);
    console.log("\n" + c.cyan(line));
    console.log("  " + c.bold(c.cyan(title)));
    if (subtitle) console.log("  " + c.gray(subtitle));
    console.log(c.cyan(line) + "\n");
  },
};

File: internal/util/logger.go
package util

import (
	"fmt"
	"os"
	"strings"
)

type logger struct {
	verbose bool
	quiet   bool
}

// L is the shared logger instance.
var L = &logger{}

func SetVerbose(v bool) {
	L.verbose = v
	if v {
		L.quiet = false
	}
}

func SetQuiet(q bool) { L.quiet = q }

func (l *logger) Info(msg string) { fmt.Println(C.Cyan(Sym.Info) + " " + msg) }
func (l *logger) Ok(msg string)   { fmt.Println(C.Green(Sym.Check) + " " + msg) }
func (l *logger) Warn(msg string) { fmt.Println(C.Yellow(Sym.Warn) + " " + msg) }
func (l *logger) Err(msg string)  { fmt.Fprintln(os.Stderr, C.Red(Sym.Cross)+" "+msg) }

func (l *logger) Step(msg string) {
	if !l.quiet {
		fmt.Println("\n" + C.Bold(C.Magenta(Sym.Arrow+" "+msg)))
	}
}

func (l *logger) Sub(msg string) {
	if !l.quiet {
		fmt.Println("  " + C.Gray(Sym.Bullet) + " " + msg)
	}
}

func (l *logger) Debug(msg string) {
	if l.verbose {
		fmt.Println("  " + C.Gray("[debug] "+msg))
	}
}

func (l *logger) Raw(msg string) { fmt.Println(msg) }

func (l *logger) Banner(title, subtitle string) {
	if l.quiet {
		return
	}
	width := len(title)
	if len(subtitle) > width {
		width = len(subtitle)
	}
	line := strings.Repeat("─", width+4)
	fmt.Println("\n" + C.Cyan(line))
	fmt.Println("  " + C.Bold(C.Cyan(title)))
	if subtitle != "" {
		fmt.Println("  " + C.Gray(subtitle))
	}
	fmt.Println(C.Cyan(line) + "\n")
}
